docs/proposals/hpa/federated-hpa-v2.md

```diff
@@ -0,0 +1,809 @@
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate them? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. This proposal doesn't consider cluster disaster migration for HPA, which should be done in another proposal.
+1. When assigning the FederatedHPA, we use availableReplicas as the key factor, it may cause inaccurate HPA assignments.
+
+## Design Details
+### Architecture
+![Karmada Federated HPA Architecture](./statics/karmadafederatedhpa-arch-v2.drawio.png)
+
+From the above architecture, FederatedHPA controller should be responsible for the following things:
+* Based on the FederatedHPA configuration, assign the minReplicas/maxReplicas of HPA to member clusters by creating works.
+* When HPA initial assignment, the controller should update the workload's ResourceBinding based on the assignment results.
+* When the member cluster could scale up the workload, FederatedHPA controller should not scale up the workload across clusters, but if the member cluster can not, FederatedHPA controller should scale the workload in other member clusters with different policies.
+* If there is a priority policy, the FederatedHPA controller should scale down the low-priority cluster first, if there is no priority policy, the FederatedHPA controller should not care about the scaling down operation.
+* The HPA resource version will be `autoscaling/v2`, because the new fields introduced in autoscaling/v2 are preserved as annotations when working with autoscaling/v1. [Referring](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)
+
+To isolate from PP/CPP and avoid PP changes that make it difficult for FederatedHPA to evolve, so we introduce FedertedHPA API to implement the scaling across clusters with rich policies.
+```go
+type FedertedHAP struct {
+    metav1.TypeMeta
+    metav1.ObjectMeta
+
+    // Spec defines the desired behavior of federated HPA.
+    Spec FederatedHPASpec `json:"spec,omitempty"`
+
+    // Status of the current information about the federated HPA.
+    // +optional
+    Status FederatedHPAStatus `json:"status,omitempty"`
+}
+
+type FederatedHPASpec struct {
+    // ScaleTargetRef points to the target resource to scale, and is used to the pods for which metrics
+    // should be collected, as well as to actually change the replica count.
+    ScaleTargetRef autoscalingv2.CrossVersionObjectReference `json:"scaleTargetRef"`
```

Comment:
`FederatedHPASpec `has all the fields of `HorizontalPodAutoscalerSpec`.
Why not inline the `HorizontalPodAutoscalerSpec` instead?

Reply:
Good idea, I will update it. Same with the status spec.

Reply:
HorizontalPodAutoscalerSpec has `minReplicas/maxReplicas`, but in FederatedHPA, it's `federatedMinReplicas/federatedMaxReplicas`, so can't inline `HorizontalPodAutoscalerSpec`,  but the status field can.